09_json/willshen8/pkg/puppy/store/mapStore.go

```diff
@@ -0,0 +1,73 @@
+package puppy
+
+import (
+	"strconv"
+
+	puppy "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+)
+
+type MapStore struct {
+	ms     map[uint32]puppy.Puppy
+	nextID uint32
+}
+
+// NewMapStore initialise a new MapStore
+func NewMapStore() *MapStore {
+	return &MapStore{ms: map[uint32]puppy.Puppy{}}
+}
+
+// CreatePuppy create a new puppy and store in mapStore.
+func (m *MapStore) CreatePuppy(p *puppy.Puppy) (uint32, error) {
+	if i, err := strconv.Atoi(p.Value); err == nil {
+		if i < 0 {
+			return 0, &puppy.Error{
+				Message: "Puppy value can't be less than 0.",
+				Code:    puppy.NegativeValue,
+			}
+		}
+	}
```

Comment:
how about the `err != nil` case?

Reply:
I guess the solution would be either change the type to int type or return a different error saying 'the value isn't recognised' or something similar. Thanks for pointing it out. And has added this condition and corresponding test case,